src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +57,592 @@ ActorInit En_Zod_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAF3E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAF4D8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAF7CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAF99C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFA44.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFADC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFB84.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFC00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFC10.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFD00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFDB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFE34.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BAFF14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/EnZod_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BB0128.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BB0170.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BB01B0.s")
+static AnimationHeader* sAnimations[] = {
+    &gTijoPlayingVivaceAnim, &gTijoReadyToPlayAnim,      &gTijoArmsFoldedAnim,
+    &gTijoPlayingLentoAnim,  &gTijoPlayingAndantinoAnim,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/EnZod_Draw.s")
+static Vec3f D_80BB0580 = { 1300.0f, 1100.0f, 0.0f };
+
+void EnZod_Init(Actor* thisx, PlayState* play) {
+    s32 i;
+    EnZod* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 60.0f);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gTijoSkel, &gTijoPlayingLentoAnim, this->morphTable, this->jointTable,
+                       TIJO_LIMB_MAX);
+    Animation_PlayLoop(&this->skelAnime, &gTijoPlayingLentoAnim);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->cymbalRots[0] = this->cymbalRots[1] = this->cymbalRots[2] = 0;
+    this->actor.gravity = this->actor.terminalVelocity = -4.0f;
+    this->cymbalRotVels[0] = this->cymbalRotVels[1] = this->cymbalRotVels[2] = 300;
+    this->stateFlags = 0;
+    this->nextAnimIndex = -1;
+    this->curAnimIndex = -1;
+    this->actor.textId = 0;
+    this->unk_298 = 0;
+
+    for (i = 0; i < ARRAY_COUNT(this->drumScales); i++) {
+        this->drumScales[i] = 0.0f;
+        this->drumScaleVels[i] = 0.01;
+    }
+
+    EnZod_ChangeAnim(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+    this->actionFunc = EnZod_PlayDrumsSequence;
+
+    switch (ENZOD_GET_TYPE(thisx)) {
+        case 1:
```

Comment:
Make an enum for the used types, as usual, and put it in the header. Looks like 1,2,0xF are used in-game.